backend/de.metas.business/src/main/java/de/metas/product/impl/ProductDAO.java

```diff
@@ -593,4 +594,34 @@ public void clearIndividualMasterDataFromProduct(final ProductId productId)
 
 		saveRecord(product);
 	}
+
+	@Override
+	public Optional<de.metas.product.model.I_M_Product> getProductOfGroupCategory(
+			@NonNull final GroupCategoryId groupCategoryId,
+			@NonNull final OrgId targetOrgId)
+	{
+
+		final ProductId targetProductId = queryBL.createQueryBuilder(I_M_Product.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_M_Product.COLUMNNAME_AD_Org_ID, targetOrgId)
+				.addEqualsFilter(I_M_Product.COLUMNNAME_C_CompensationGroup_Schema_Category_ID, groupCategoryId)
+				.orderByDescending(I_M_Product.COLUMNNAME_M_Product_ID)
+				.create()
+				.firstId(ProductId::ofRepoIdOrNull);
+
+		if (targetProductId == null)
+		{
+			return Optional.empty();
+		}
+
+		return Optional.of(getById(targetProductId, de.metas.product.model.I_M_Product.class));
+	}
+
+	public Optional<GroupCategoryId> getGroupCategoryId(@NonNull final ProductId productId)
+	{
+		final de.metas.product.model.I_M_Product productRecord = getById(productId, de.metas.product.model.I_M_Product.class);
+		final int groupCategoryId = productRecord.getC_CompensationGroup_Schema_Category_ID();
+		return Optional.of(GroupCategoryId.ofRepoIdOrNull(groupCategoryId));
```

Comment:
use Optional.ofNullable

Reply:
I'm not using this function anymore because I use de.metas.contracts.bpartner.repository.OrgChangeRepository#getGroupCategoryId.

Deleting it